Foundation/CPCache.j

```diff
@@ -0,0 +1,350 @@
+    }
+
+    [_items removeAllObjects];
+}
+
+
+#pragma mark Accessors
+
+/*!
+    Returns the count limit of the cache
+*/
+- (int)countLimit
+{
+    return _countLimit;
+}
+
+/*!
+    Sets the count limit of the cache.
+    Remove objects if not enough place to keep all of them
+    @param aCountLimit the new count limit
+*/
+- (void)setCountLimit:(int)aCountLimit
+{
+    _countLimit = aCountLimit;
+    [self _cleanCache];
+}
+
+/*!
+    Returns the total cost limit of the cache
+*/
+- (int)totalCostLimit
+{
+    return _totalCostLimit;
+}
+
+/*!
+    Sets the total cost limit of the cache.
+    Remove objects if not enough place to keep all of them
+    @param aTotalCostLimit the new total cost limit
+*/
+- (void)setTotalCostLimit:(int)aTotalCostLimit
+{
+    _totalCostLimit = aTotalCostLimit;
+    [self _cleanCache];
+}
+
+/*!
+    Returns the cache's delegate
+*/
+- (id)delegate
+{
+    return _delegate;
+}
+
+/*!
+    Sets the cache's delegate.
+    @param aDelegate the new delegate
+*/
+- (void)setDelegate:(id)aDelegate
+{
+    if (_delegate === aDelegate)
+        return;
+
+    _delegate = aDelegate;
+    _implementedDelegateMethods = 0;
+
+    if ([_delegate respondsToSelector:@selector(cache:willEvictObject:)])
+        _implementedDelegateMethods |= CPCacheDelegate_cache_WillEvictObject
+}
+
+
+#pragma mark Privates
+
+/*
+ * This method return the number of objects in the cache
+ */
+- (int)_count
+{
+    return [_items count];
+}
+
+/*
+ * This method return the total cost (addition of all object's cost in the cache)
+ */
+- (int)_totalCost
+{
+    var enumerator = [_items objectEnumerator],
+        cost = 0,
+        value;
+
+    while (value = [enumerator nextObject])
+        cost += [value cost];
+
+    return cost;
+}
+
+/*
+ * This method resequence the position of objects
+ * Otherwise the position value could rise until to cause problem
+ */
+- (void)_resequencePosition
+{
+    _currentPosition = 1;
+
+    // Sort keys by position
+    var sortedKeys = [[_items allKeys] sortedArrayUsingFunction:function(k1, k2, context)
+    {
+        var o1 = [_items objectForKey:k1],
+            o2 = [_items objectForKey:k2];
+        return ([o1 position] < [o2 position] ? CPOrderedAscending : ([o1 position] > [o2 position] ? CPOrderedDescending : CPOrderedSame));
+    } context:nil];
+
+    // Affect new positions
+    for (var i = 0; i < sortedKeys.length; ++i)
+        [[_items objectForKey:sortedKeys[i]] setPosition:_currentPosition++];
+}
+
+/*
+ * This method clean the cache if the totalCost or the count exceeds the totalCostLimit or the countLimit
+ * until to satisfy conditions (totalCost < totalCostLimit and count < countLimit)
+ */
+- (void)_cleanCache
+{
+    // Check if the condition is satisfied (totalCost < totalCostLimit and count < countLimit)
+    if (([self _totalCost] > _totalCostLimit && _totalCostLimit > 0) || ([self _count] > _countLimit && _countLimit > 0))
```

Comment:
we usually filter out this conditions that are "if that don't do anything"

you should do a reverse if, and if it matches, you simply `return`